CONTRIBUTING.md

```diff
@@ -128,9 +128,21 @@ You can also configure this task to execute via a keyboard shortcut by [configur
 }
 ```
 
-### Code owners
+### Code review
 
-Each file in a pull request requires the approval of at least one of its code owners (though in general for interesting changes we wait for everyone to review). Owners for different files are listed in [`CODEOWNERS`](/.github/CODEOWNERS).
+This repo follows [the NI code submission workflow](https://ni.visualstudio.com/DevCentral/_wiki/wikis/AppCentral.wiki/15679/Code-Submission-Workflow#) (NI internal link) with some modifications. The general submission flow is:
+
+1. When your code is ready to submit, create a [Draft pull request](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#draft-pull-requests).
+2. Perform a self review of the changes. Ensure they follow conventions and that the status checks are passing.
+3. Add a local peer reviewer. If you are new to Nimble, also add the contact from the Nimble team who has been helping you get familiar with the repo.
+4. Once those reviewers have approved, mark the PR as "Ready for review". This will add owners to the review.
+5. Each file in a pull request requires the approval of at least one of its code owners (though in general for interesting changes we wait for everyone to review). Owners for different files are listed in [`CODEOWNERS`](/.github/CODEOWNERS).
+
+If a PR requires significant refactoring at any point in this process, please move it back to Draft and re-do the steps before adding owners again.
```

Comment:
Once owners (as defined by CODEOWNERS) are added I don't think can be removed even if moved back to draft. It's a slightly annoying minor nit as folks often accidentally create the normal PR and switch to draft so get the notifications anyway.

```suggestion
If a PR requires significant refactoring at any point in this process, please move it back to Draft and re-do the steps before exiting draft.
```